vars/transportRequestCreate.groovy

```diff
@@ -37,7 +45,33 @@ def call(parameters = [:]) {
                                                       stepConfigurationKeys)
 
         def changeDocumentId = configuration.changeDocumentId
-        if(!changeDocumentId) throw new AbortException('Change document id not provided (parameter: \'changeDocumentId\').')
+
+        if(changeDocumentId?.trim()) {
+
+            echo "[INFO] ChangeDocumentId '${changeDocumentId}' retrieved from parameters."
+
+        } else {
+
+            echo "[INFO] Retrieving ChangeDocumentId from commit history [from: ${configuration.gitFrom}, to: ${configuration.gitTo}]." +
+                 "Searching for pattern '${configuration.gitChangeDocumentLabel}'. Searching with format '${configuration.gitFormat}'."
+
+            try {
+                changeDocumentId = cm.getChangeDocumentId(
+                                                          configuration.gitFrom,
+                                                          configuration.gitTo,
+                                                          configuration.gitChangeDocumentLabel,
+                                                          configuration.gitFormat
+                                                         )
+
+                echo "[INFO] ChangeDocumentId '${changeDocumentId}' retrieved from commit history"
+            } catch(ChangeManagementException ex) {
+                echo "[WARN] Cannot retrieve changeDocumentId from commit history: ${ex.getMessage()}."
```

Comment:
why dont you exit here?

Reply:
Would also be possible. I decided to separate retrieval of parameters and validation of parameters. Here we are in the retrieval phase, so we only retrieve. Later we have some lines performing the validation.
In the long run I would like to use the validation facilities provided by the configuration helper. See PR #202